src/content/docs/apm/agents/go-agent/configuration/go-agent-code-level-metrics-config.mdx

```diff
@@ -0,0 +1,286 @@
+---
+title: Go agent code-level metrics configuration
+tags:
+  - Agents
+  - Go agent
+  - Configuration
+metaDescription: Add source code context to your application traces, associating the trace data with the specific code being instrumented.
+---
+
+You can edit configuration settings for the Go agent to control the collection of code-level metrics. For general information about code-level metrics, see [Performance monitoring with CodeStream](/docs/codestream/how-use-codestream/performance-monitoring#code-level).
+
+## Overview [#overview]
+
+With code-level metrics enabled, the Go agent will attach attributes to trace data which give the location in your application source code
+responsible for the actions being instrumented by those traces. This data includes:
+ * Source file name
+ * Source file line number
+ * Function name
+ * Namespace
+
+<Callout variant="important">
+    This feature is available for transactions as of Go agent version 3.18.0, and is turned off by default. To enable it, you must
+    add `newrelic.ConfigCodeLevelMetricsEnabled(true)` to your application's configuration as explained below. Future releases may
+    change this to be on by default and may expand the feature to be included with other trace types other than transactions.
+</Callout>
```

Comment:
This callout is bigger than the preceding paragraphs. The result is an awkward tone. It features a slightly negative callout dominating the space above the fold. It would be nice if you could enhance the opening paragraphs by even just a sentence to make the feature the focal point, not the callout.

Reply:
@homelessbirds, I removed that final sentence in the callout since it talks about an indefinite future state. So, this callout is shorter.